utils/node_local_test.py

```diff
@@ -1248,6 +1249,11 @@ def get_cmd_prefix(self):
         if not self.use_valgrind:
             return []
 
+        # valgrind reduces the hard limit unless we bump the soft limit first
+        (soft, hard) = resource.getrlimit(resource.RLIMIT_NOFILE)
+        if soft < hard:
+            resource.setrlimit(resource.RLIMIT_NOFILE, (hard, hard))
+
```

Comment:
This would be better set after passing the command line at https://github.com/daos-stack/daos/blob/master/utils/node_local_test.py#L6530